internal/pkg/template/workload.go

```diff
@@ -540,8 +540,9 @@ type DeadLetterQueue struct {
 
 // NetworkOpts holds AWS networking configuration for the workloads.
 type NetworkOpts struct {
-	SecurityGroups []string
-	AssignPublicIP string
+	PlainSecurityGroups    []string
+	ImportedSecurityGroups []string
```

Comment:
what do you think of doing something similar with `Variable`? https://github.com/aws/copilot-cli/blob/e5c44581e69659a874f5d7ebadd7119dc551309e/internal/pkg/template/workload.go#L263

For example,

```go
type NetworkOpts struct {
  SecurityGroups []SecurityGroup
}

type SecurityGroup interface {
  importable
  Value()
}

type PlainSecurityGroup string

func (sg PlainSecurityGroup) RequiresImport() bool {
  return false
}

type ImportedSecurityGroup string

func (sg ImportedSecurityGroup) RequiresImport() bool {
  return true
}
```